### PR TITLE
注册实例时支持指定实例id

### PR DESCRIPTION
**Pull request by movebean**
**Please provide issue(s) of this PR:**
#126

**To help us figure out who should review this PR, please put an X in all the areas that this PR affects.**

- [ ] Configuration
- [ ] Docs
- [ ] Performance and Scalability
- [X ] Naming
- [ ] HealthCheck
- [ ] Test and Release

**Please check any characteristics that apply to this pull request.**

- [ ] Does not have any user-facing changes. This may include API changes, behavior changes, performance improvements, etc.

### CODE REVIEW

File: pkg/model/service.go [hunk 1]
```diff
@@ -1217,6 +1217,8 @@ type InstanceRegisterRequest struct {
 	Timeout *time.Duration
 	// 可选，重试次数，默认直接获取全局的超时配置
 	RetryCount *int
+	// 可选，指定实例id
+	InstanceId string
```
Comment (houseme) [inline pkg/model/service.go:1221]: 应该和上面的一样，统一为指针类型

Comment (andrewshan) [reply pkg/model/service.go:1221]: 不需要，空串和空指针的意义是一样的，都代表没有指定。
上面2个为0和为空的含义不一样

File: examples/quickstart/provider/main.go [hunk 2]
```diff
@@ -99,6 +99,7 @@ func (svr *PolarisProvider) registerService() {
 	registerRequest.Port = svr.port
 	registerRequest.ServiceToken = token
 	registerRequest.SetTTL(10)
+	registerRequest.InstanceId = "instance-id-provided"
```
Comment (andrewshan) [inline examples/quickstart/provider/main.go:102]: 这里可以加个注释，说明instanceID不是必填的，如果不填入，则服务端会默认生成一个唯一的ID，如果填了，则必须保证唯一性

Comment (movebean) [reply examples/quickstart/provider/main.go:103]: 加上了注释标明实例id不是必填，如果填入，必须保证唯一

**Review by andrewshan — APPROVED** [review]
LGTM

**Pull request merged**